Podcast/SearchViewController.swift

```diff
@@ -0,0 +1,151 @@
+        pastSearchesTableView.dataSource = self
+        pastSearchesTableView.separatorStyle = .none
+        view.addSubview(pastSearchesTableView)
+        
+        pastSearchesTableView.snp.makeConstraints { make in
+            make.width.equalToSuperview()
+            make.top.equalToSuperview()
+            make.bottom.equalToSuperview().inset(appDelegate.tabBarController.tabBarHeight)
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        if let priorSearches = UserDefaults.standard.value(forKey: "PastSearches") as? [String] {
+            pastSearches = Set(priorSearches)
+        } else {
+            pastSearches = nil 
+        }
+        reloadPastSearchTableViewData()
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        searchController?.searchBar.isHidden = true
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        searchController?.searchBar.isHidden = false
+    }
+    
+    //MARK: - Tabbed Search Results Delegate
+    func didTapOnSeriesCell(series: Series) {
+        addPastSearches()
+        let seriesDetailViewController = SeriesDetailViewController(series: series)
+        navigationController?.pushViewController(seriesDetailViewController,animated: true)
+    }
+    
+    func didTapOnEpisodeCell(episode: Episode) {
+        addPastSearches()
+        let episodeViewController = EpisodeDetailViewController()
+        episodeViewController.episode = episode
+        navigationController?.pushViewController(episodeViewController, animated: true)
+    }
+    
+    func didTapOnUserCell(user: User) {
+        addPastSearches()
+        let externalProfileViewController = ExternalProfileViewController()
+        externalProfileViewController.fetchUser(id: user.id)
+        navigationController?.pushViewController(externalProfileViewController, animated: true)
+    }
+    
+    func addPastSearches() {
+        guard let searchText = searchController.searchBar.text else { return }
+        if searchText == "" { return }
+        if var searches = pastSearches {
+            if !(searches.contains(searchText)) {
+                searches.insert(searchText)
+                UserDefaults.standard.set(Array(searches), forKey: "PastSearches")
+            }
+        } else {
+            UserDefaults.standard.set([searchText], forKey: "PastSearches")
+        }
+    }
+    
+    //MARK: -
+    //MARK: UITableViewDelegate & Data source
+    //MARK: -
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "PastSearchCell") as? PastSearchTableViewCell, let priorSearches = UserDefaults.standard.value(forKey: "PastSearches") as? [String]  else { return UITableViewCell() }
+        cell.label.text = priorSearches[priorSearches.count - indexPath.row - 1] //searches are stored in reverse order due to nature of appending to array
```

Comment:
Why not use queue/linked-list?

Reply:
Overhead i'd say of creating node structures and saving that to user defaults .. also didn't want to add to the beginning of the array because that will shift items every time! but maybe adding to the end actually copies the items to a new array because it has to increase size. Let me look into this.

Reply:
LinkedLists are not part of standard swift library, queues are just arrays really and i'm not trying to remove elements just efficiently adding something to the beginning. I am going to to do insert at index 0 for now and remove the reversing during cellForRowAt. If we find this is inefficient we can change it later